apps/files/src/components/LocationPicker/LocationPicker.vue

```diff
@@ -0,0 +1,369 @@
+        })
+      }
+
+      if (this.target) {
+        const items = this.target.split('/').filter(item => item !== '')
+
+        for (let i = 0; i < items.length; i++) {
+          const itemPath = encodeURIComponent(pathUtil.join.apply(null, items.slice(0, i + 1)))
+
+          breadcrumbs.push({
+            index: i + 1,
+            text: items[i],
+            to: this.createPath(itemPath)
+          })
+        }
+      }
+
+      delete breadcrumbs[breadcrumbs.length - 1].to
+
+      return breadcrumbs
+    },
+
+    canConfirm() {
+      return this.currentFolder?.canCreate()
+    },
+
+    title() {
+      const count = this.resourcesCount
+
+      if (this.currentAction === 'move') {
+        const title = this.$ngettext(
+          'Selected %{ count } resource to move into:',
+          'Selected %{ count } resources to move into:',
+          count
+        )
+
+        return this.$gettextInterpolate(title, { count: count })
+      }
+
+      return null
+    },
+
+    confirmBtnText() {
+      if (this.currentAction === 'move') {
+        return this.$gettext('Move here')
+      }
+
+      return null
+    }
+  },
+
+  watch: {
+    $route: {
+      handler: 'navigateToTarget',
+      immediate: true
+    }
+  },
+
+  created() {
+    this.originalLocation = this.target
+
+    if (this.currentAction === 'move') {
+      this.SET_MAIN_CONTENT_COMPONENT(MoveSidebarMainContent)
+    }
+  },
+
+  beforeDestroy() {
+    this.SET_MAIN_CONTENT_COMPONENT(null)
+  },
+
+  methods: {
+    ...mapMutations(['SET_MAIN_CONTENT_COMPONENT']),
+    ...mapActions('Files', ['loadFolder']),
+    ...mapActions(['showMessage']),
+
+    navigateToTarget(target) {
+      if (typeof target === 'object') {
+        target = this.target
+      }
+
+      this.loadFolder({
+        client: this.$client,
+        absolutePath: target || '/',
+        $gettext: this.$gettext,
+        routeName: this.$route.name,
+        loadSharesTree: !this.isPublicPage,
+        isPublicPage: this.isPublicPage
+      })
+    },
+
+    createPath(target) {
+      return (
+        this.basePath +
+        `?action=${this.currentAction}&target=` +
+        encodeURIComponent(target) +
+        this.resourcesQuery
+      )
+    },
+
+    selectFolder(folder) {
+      if (this.isRowDisabled(folder)) {
+        return
+      }
+
+      this.$router.push({ path: this.createPath(folder.path) })
+    },
+
+    leaveLocationPicker(target) {
+      if (this.isPublicPage) {
+        this.$router.push({ name: 'public-link', params: { token: target } })
+
+        return
+      }
+
+      this.$router.push({ name: 'files-list', params: { item: target } })
+    },
+
+    async moveResources() {
+      const errors = []
+      const promise = this.isPublicPage ? this.$client.publicFiles : this.$client.files
+
+      for (const resource of this.resources) {
+        let target = this.target || '/'
+        const resourceName = pathUtil.basename(resource)
```

Comment:
I know that you just changed to using the `pathUtil.basename`. But one more remark on this: in `PublicLinkListItem.vue` and `Collaborator.vue` we were using the `basename` function like this:
`import { basename, dirname } from 'path'` and then just `basename(...)`. Would be nice to have consistent usages.